### PR TITLE
Release code review: grep design and implementation

### PR DESCRIPTION
**Pull request by pearlkwok**
https://www.notion.so/Grep-Demo-App-Usage-9438c3eda2bb43939f2056cddd643516
Something need to fix:
sl4fj not able to import

### CODE REVIEW

**Comment by pearlkwok** [comment]
https://www.notion.so/Grep-Demo-App-Usage-9438c3eda2bb43939f2056cddd643516
Something need to be fixed:
cannot import slf4j
cannot import junit
write junit tests

File: core_java/.idea/core_java.iml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<module type="JAVA_MODULE" version="4">
+  <component name="NewModuleRootManager" inherit-compiler-output="true">
+    <exclude-output />
+    <content url="file://$MODULE_DIR$" />
```
Comment (jarviscanada) [inline core_java/.idea/core_java.iml:5]: git rm .idea

remove all generated files

File: core_java/grep/sampleOutput [hunk 2]
```diff
@@ -0,0 +1,5 @@
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
```
Comment (jarviscanada) [inline core_java/grep/sampleOutput:2]: remove this too

File: core_java/grep/src/main/java/ca/jrvs/apps/grep/JavaGrepImp.java [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+//import org.slf4j.Logger;
+//import org.slf4j.LoggerFactory;
+
+import java.io.*;
+import java.util.ArrayList;
+import java.util.List;
+
+public class JavaGrepImp implements JavaGrep {
+
+    //final Logger logger = LoggerFactory.getLogger(JavaGrep.class);
```
Comment (jarviscanada) [inline core_java/grep/src/main/java/ca/jrvs/apps/grep/JavaGrepImp.java:13]: clean up your comments. you still need a logger

File: core_java/grep/src/main/java/ca/jrvs/apps/practice/HelloWorld.java [hunk 4]
```diff
@@ -0,0 +1,24 @@
+
+package ca.jrvs.apps.practice;
+
+class HelloWorld {
+
+  // Your program begins with a call to main().
+  //   // Prints "Hello, World" to the terminal window.
+	public static void main(String args[]) {
```
Comment (jarviscanada) [inline core_java/grep/src/main/java/ca/jrvs/apps/practice/HelloWorld.java:8]: remove

File: core_java/grep/target/maven-archiver/pom.properties [hunk 5]
```diff
@@ -0,0 +1,5 @@
+#Generated by Maven
+#Mon Apr 06 18:03:50 UTC 2020
```
Comment (jarviscanada) [inline core_java/grep/target/maven-archiver/pom.properties:2]: git rm target 

remove all `target` dir

**Review by jarviscanada — APPROVED** [review]

**Pull request merged**